### PR TITLE
Data Visualization: Add usage guidelines with examples

### PR DESCRIPTION
**Pull request by ayeshakmaz**
#### What changed?

Add page for examples and guidelines around the new data visualization color palette

#### Why?

Provide examples of color pairings, and pairings to avoid 

### Links

- [Figma](https://www.figma.com/file/UOCLenizykiIi9yxs8k992/Working-file%3A-Data-viz-palette?node-id=834%3A27996)

### Checklist

- [x] Added unit and Flow Tests
- [x] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/pages/data_visualization_guidelines.js [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+type ColorCardProps = {|
+  count: number,
+|};
+function PaletteGenerator({ count }: ColorCardProps): Node {
+  return [...Array(count)].map((step, idx) => {
+    const tokenStep = idx + 1;
+    return (
+      <Box marginBottom={1} key={`color-${idx}`}>
+        <ColorTile
+          key={`${idx}`}
```
Comment (dangerismycat) [inline docs/pages/data_visualization_guidelines.js:25]: this key isn't doing anything

File: docs/pages/data_visualization_guidelines.js [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+                fullTokenName="color-data-visualization-05"
+              />
+            </Flex>
+            <Flex direction="column" gap={1}>
+              <ColorTile
+                textColor="inverse"
+                description="Data Visualization 08"
+                fullTokenName="color-data-visualization-08"
+              />
+              <Box marginBottom={6}>
```
Comment (dangerismycat) [inline docs/pages/data_visualization_guidelines.js:201]: is this marginBottom necessary?

**Review by dangerismycat — APPROVED** [review]

File: docs/pages/data_visualization_guidelines.js (hunk 2 above)
Comment (ayeshakmaz) [reply docs/pages/data_visualization_guidelines.js:201]: yeah it's for when it wraps into one column on mobile. Not the cleanest solution, but I'll add a comment

File: docs/pages/data_visualization_guidelines.js (hunk 1 above)
Comment (ayeshakmaz) [reply docs/pages/data_visualization_guidelines.js:25]: good catch!

File: docs/pages/data_visualization_guidelines.js (hunk 2 above)
Comment (dangerismycat) [reply docs/pages/data_visualization_guidelines.js:201]: 👍 This is why I need to re-implement `gap` to include both column and row values 😬

Comment (ayeshakmaz) [reply docs/pages/data_visualization_guidelines.js:201]: Yes please :P

**Pull request merged**